fedimint-api-client/src/api/mod.rs

```diff
@@ -1046,28 +1003,252 @@ impl IClientConnection for Arc<WsClient> {
+    /// Create a builder with recommended defaults intended for client-side
+    /// usage
+    ///
+    /// In particular mobile devices are considered.
+    pub fn build_from_client_defaults() -> EndpointRegistryBuilder {
+        EndpointRegistryBuilder {
+            iroh_enable: true,
+            iroh_dns: None,
+            iroh_pkarr_dht: false,
+            iroh_next: true,
+            ws_enable: true,
+            ws_force_tor: false,
+
+            connection_overrides: BTreeMap::default(),
+        }
+    }
+
+    /// Create a builder with recommended defaults intended for the server-side
+    /// usage
+    pub fn build_from_server_defaults() -> EndpointRegistryBuilder {
+        EndpointRegistryBuilder {
+            iroh_enable: true,
+            iroh_dns: None,
+            iroh_pkarr_dht: true,
+            iroh_next: true,
+            ws_enable: true,
+            ws_force_tor: false,
+
+            connection_overrides: BTreeMap::default(),
+        }
+    }
+
+    /// Create a builder with recommended defaults intended for testing
+    /// usage
+    pub fn build_from_testing_defaults() -> EndpointRegistryBuilder {
+        EndpointRegistryBuilder {
+            iroh_enable: true,
+            iroh_dns: None,
+            iroh_pkarr_dht: false,
+            iroh_next: false,
+            ws_enable: true,
+            ws_force_tor: false,
+
+            connection_overrides: BTreeMap::default(),
+        }
+    }
+
+    /// Like [`Self::build_from_client_defaults`] build will apply
+    /// environment-provided overrides.
+    pub fn build_from_client_env() -> anyhow::Result<EndpointRegistryBuilder> {
+        let builder = Self::build_from_client_defaults().with_env_var_overrides()?;
+        Ok(builder)
+    }
+
+    /// Like [`Self::build_from_server_defaults`] build will apply
+    /// environment-provided overrides.
+    pub fn build_from_server_env() -> anyhow::Result<EndpointRegistryBuilder> {
+        let builder = Self::build_from_server_defaults().with_env_var_overrides()?;
+        Ok(builder)
+    }
+
+    /// Like [`Self::build_from_testing_defaults`] build will apply
+    /// environment-provided overrides.
+    pub fn build_from_testing_env() -> anyhow::Result<EndpointRegistryBuilder> {
+        let builder = Self::build_from_testing_defaults().with_env_var_overrides()?;
+        Ok(builder)
+    }
+
+    /// Connect to a given `url` using matching [`Endpoint`]
+    ///
+    /// This is the main function consumed by the downstream use for making
+    /// connection.
+    pub async fn connect(
+        &self,
+        url: &SafeUrl,
+        api_secret: Option<&str>,
+    ) -> PeerResult<DynClientConnection> {
+        let Some(endpoint) = self.inner.get(url.scheme()) else {
+            return Err(PeerError::InvalidEndpoint(anyhow!(
+                "Unsupported scheme: {}; missing endpoint handler",
+                url.scheme()
+            )));
+        };
+
+        endpoint.connect(url, api_secret).await
```

Comment:
If we only match by scheme we back ourselves into a corner of having to establish a scheme for Tor. Now instead of `ws://dfkghdsjkgfadfkgj.onion` it would have to be something along the lines of `tor+ws://dfkghdsjkgfadfkgj.onion`. This is at least unusual.